optuna/experimental_warning.py

```diff
@@ -0,0 +1,42 @@
+import warnings
+
+
+# Reference: https://github.com/cupy/cupy/blob/8d923719ced31fdff4a42702d395ec87a7e98045/cupy/util.pyx#L89  # NOQA
+def experimental(api_name: str) -> None:
```

Comment:
Providing this as a decorator is an alternative as well, in which case the current name makes sense. We could then also skip the parameter and derive the name. The downside of creating a decorator would be that it'd be less straight forward to place the warning in e.g. a conditioned statement (but possible by encapsulating logic in new functions).

Reply:
Thanks for the suggestion. Thought the naming is terrible, but I wrote decorators tentatively for smoother discussion (I hope).